### PR TITLE
Update E2ETest to use ReactApplication

### PR DESCRIPTION
**Pull request by ddalp**
Update E2ETest to use Microsoft ReactNative application model, in preparation to be able to add Tree Dump native module to UIElement tree comparison.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3715)

### CODE REVIEW

File: packages/E2ETest/windows/ReactUWPTestApp/App.xaml.cs [hunk 1]
```diff
@@ -7,87 +7,53 @@
...
+        /// </summary>
+        /// <param name="e">Details about the launch request and process.</param>
+        protected override void OnLaunched(LaunchActivatedEventArgs e)
         {
-          // When the navigation stack isn't restored navigate to the first page,
-          // configuring the new page by passing required information as a navigation
-          // parameter
-          rootFrame.Navigate(typeof(ReactTestPage), e.Arguments);
+            base.OnLaunched(e);
+            SystemNavigationManager.GetForCurrentView().AppViewBackButtonVisibility = AppViewBackButtonVisibility.Collapsed;
```
Comment (licanhua) [inline packages/E2ETest/windows/ReactUWPTestApp/App.xaml.cs]: >AppViewBackButtonVisibility [](start = 56, length = 27)

why do we need this? Is it a bug of ReactNative

Comment (ddalp) [reply packages/E2ETest/windows/ReactUWPTestApp/App.xaml.cs]: Yes, it is not implemented yet and will throw exception when user click on back button, see ReactInstanceManager::OnBackPressed

Comment (licanhua) [reply packages/E2ETest/windows/ReactUWPTestApp/App.xaml.cs]: I suggest you create a bug to track this.

Comment (ddalp) [reply packages/E2ETest/windows/ReactUWPTestApp/App.xaml.cs]: Spoke with Jon and created issue #3721  to track.

**Pull request merged**